sap/cli/checkin.py

```diff
@@ -415,13 +415,154 @@ def _check_fugr_source_files(repo_obj, functions, includes):
             raise sap.adt.errors.ExceptionCheckinFailure(f'No source file for include {include_name}')
 
 
+def _write_source_file(source_code, adt_object, corrnr=None):
+    with adt_object.open_editor(corrnr=corrnr) as editor:
+        editor.write(source_code)
+
+
 def _write_adt_object_source_file(path_prefix, adt_object, corrnr=None):
     """Write source file for ADT object"""
 
     adt_object_file_path = path_prefix + f'.{adt_object.name.lower()}' + '.abap'
     with open(adt_object_file_path, 'r', encoding='utf-8') as source:
-        with adt_object.open_editor(corrnr=corrnr) as editor:
-            editor.write(source.read())
+        _write_source_file(source.read(), adt_object, corrnr)
+
+
+def _get_parameters_block(source_lines):
+    """Get parameters block of function module
+
+    The block is delimited by '*"--' lines.
+    Example:
+    *"----------------------------------------------------------------------
+    *"*"Local Interface:
+    ...
+    *"----------------------------------------------------------------------
+    """
+
+    start_block = 0
+    end_block = 0
+    for i, line in enumerate(source_lines):
+        if line.startswith('*"--'):
+            start_block = i
+            break
+
+    for i, line in enumerate(source_lines[start_block + 1:]):
+        if line.startswith('*"--'):
+            end_block = i + start_block + 1
+            break
+
+    return start_block, end_block
+
+
+def _parse_function_parameters(parameters_block):
+    """Parse parameters of function module
+
+    From the parameters block:
+    ```
+        *"----------------------------------------------------------------------
+        *"*"Local Interface:
+        *"  IMPORTING
+        *"     VALUE(IV_PARAM1) TYPE  STRING
+        *"  EXPORTING
+        *"     VALUE(EV_PARAM2) TYPE  STRING
+        *"  TABLES
+        *"     ET_PARAM3 STRUCTURE  STRING
+        *"----------------------------------------------------------------------
+    ```
+    The parsed parameters are:
+    {
+        'IMPORTING': ['VALUE(IV_PARAM1) TYPE STRING'],
+        'EXPORTING': ['VALUE(EV_PARAM2) TYPE STRING'],
+        'CHANGING': [],
+        'TABLES': ['ET_PARAM3 TYPE  STRING'],
+        'EXCEPTIONS': []
+    }
+
+    Note the change from STRUCTURE to TYPE for TABLES parameters.
+    """
+
+    parameters = {
+        'IMPORTING': [],
+        'EXPORTING': [],
+        'CHANGING': [],
+        'TABLES': [],
+        'EXCEPTIONS': []
+    }
+    current_param = None
+    for line in parameters_block:
+        if any(param in line for param in parameters):
```

Comment:
what if I have a parameter named "READ_TABLES"?

Reply:
I would use a regex or simply always do `param = line.lstrip('*" ').rstrip()` and use the equality operator for checking if the current line is a "param type" label.